### PR TITLE
adpcm-decoder fix parsing incorrectly

### PR DESCRIPTION
**Pull request by chrisderwahre**
This fixes that the adpcm-decoder creates invalid files, it parses the chunk size at an invalid offset, which results in invalid output files.

Tested on Windows 10

### CODE REVIEW

File: adpcm-decoder/main.c [hunk 1]
```diff
@@ -14,14 +15,34 @@ int main(int argc, char* argv[]) {
...
-  fseek(in, 20, SEEK_SET);
+  uint32_t riff_size;
+  fread(&riff_size, 4, 1, in);
+  printf("RIFF size: %d\n", riff_size);
 
-  // Parse 'fmt ' chunk
+  char wave_id[4];
+  fread(wave_id, 4, 1, in);
+  if (strncmp(wave_id, "WAVE", 4) != 0) {
+    fprintf(stderr, "Invalid WAVE Id: %s\n", wave_id);
```
Comment (JayFoxRox) [inline adpcm-decoder/main.c]: Is this an "ID" (consider all caps?) or a magic?
(I'd have to check RIFF / WAVE specs)

Comment (chrisderwahre) [reply adpcm-decoder/main.c]: i think it is more like a magic.

File: adpcm-decoder/main.c [hunk 2]
```diff
@@ -14,14 +15,34 @@ int main(int argc, char* argv[]) {
...
+    return 1;
+  }
+
+  char fmt_id[4];
+  fread(fmt_id, 4, 1, in);
+  if (strncmp(fmt_id, "fmt ", 4) != 0) {
+    fprintf(stderr, "Invalid fmt Id: %s\n", fmt_id);
```
Comment (JayFoxRox) [inline adpcm-decoder/main.c]: I believe this is a chunk? So it should probably say something like: "Unexpected chunk: '%.4s'. Expected 'fmt '\n" or something.

Make sure you use `%.4s` so the text is cut-off after 4 symbols

File: adpcm-decoder/main.c [hunk 3]
```diff
@@ -96,16 +96,16 @@ int main(int argc, char* argv[]) {
...
   //FIXME: Skip possible fact headers etc
 
-  char data_id[4];
-  fread(data_id, 4, 1, in);
-  if (strncmp(data_id, "data", 4) != 0) {
-    printf("Invalid DATA Id: %s\n", data_id);
+  char data_chunk_id[4];
+  fread(data_chunk_id, 4, 1, in);
+  if (strncmp(data_chunk_id, "data", 4) != 0) {
+    printf("Invalid DATA Id: %s\n", data_chunk_id);
```
Comment (JayFoxRox) [inline adpcm-decoder/main.c]: Should use the same text as the other chunk checks (also `%.4s` - obviously my original code was bad).

Hence my suggestion of moving it all in a function (but it feels like opening a can of worms; so just fix this text for now)

Please squash-up, too.

Comment (chrisderwahre) [reply adpcm-decoder/main.c]: i'll change that an squash it later!

File: adpcm-decoder/main.c [hunk 4]
```diff
@@ -14,14 +15,34 @@ int main(int argc, char* argv[]) {
...
+  char fmt_chunk_id[4];
+  fread(fmt_chunk_id, 4, 1, in);
+  if (strncmp(fmt_chunk_id, "fmt ", 4) != 0) {
+    fprintf(stderr, "Unexpected chunk: '%.4s'. Expected 'fmt '\n", fmt_chunk_id);
+    return 1;
+  }
+
+  uint32_t fmt_chunk_size;
+  fread(&fmt_chunk_size, 4, 1, in);
+  printf("Chunk size: %d\n", fmt_chunk_size);
```
Comment (JayFoxRox) [inline adpcm-decoder/main.c]: This is still inconsistent. The others are "RIFF size" and "Data size"; so why isn't this one "fmt size" or "Format size"?

**Pull request merged**